extensions/wikibase/src/org/openrefine/wikibase/editing/EditBatchProcessor.java

```diff
@@ -183,14 +185,22 @@ public EditResult performEdit()
                 // New entities
                 ReconEntityIdValue newCell = (ReconEntityIdValue) update.getEntityId();
                 EntityIdValue createdDocId;
+                String label = "";
                 if (update instanceof MediaInfoEdit) {
                     MediaFileUtils mediaFileUtils = new MediaFileUtils(connection);
                     createdDocId = ((MediaInfoEdit) update).uploadNewFile(editor, mediaFileUtils, summary, tags, filePageWaitTime,
                             filePageMaxWaitTime);
+                    label = "File:".concat(((MediaInfoEdit) update).getFileName());
                 } else {
-                    createdDocId = editor.createEntityDocument(update.toNewEntity(), summary, tags).getEntityId();
+                    LabeledDocument labeledDocument = (LabeledDocument) editor.createEntityDocument(update.toNewEntity(), summary, tags);
+                    createdDocId = labeledDocument.getEntityId();
+                    if (labeledDocument != null && labeledDocument.getLabels() != null) {
+                        label = labeledDocument.getLabels().get(Locale.getDefault().getLanguage()) != null
+                                ? labeledDocument.getLabels().get(Locale.getDefault().getLanguage()).getText()
```

Comment:
I think it would be worth having a fall back on taking *any* label if we can't find one for the current locale. And then fall back on the entity id if there isn't any label, because using the empty string as a name for a reconciliation candidate is likely to be not so pleasant for the user.